enzyme/Enzyme/GradientUtils.h

```diff
@@ -79,6 +79,13 @@ extern llvm::StringMap<std::function<llvm::Value *(
     shadowHandlers;
 
 class DiffeGradientUtils;
+
+/// This is the main entry point to register a custom derivative for language frontends.
+/// It is more general, and therefore recommended over trying to register 
+/// custom-derivatives in the llvm-ir module. Examples on why it is more general include custom-derivatives
+/// for non-default activity cases (e.g. a function call where a pointer or a float scalar is marked as const).
```

Comment:
oh can you actually write out an example here where that function call is handled using the gutils api? I think that would be helpful for people

Reply:
For simplicity I've done the example for fwd-mode only, but I can check for the reverse mode.
I've also added what Enzyme generates on it's own as a reference, since that's what we want to generate, but
it makes the example larger. I think it can help people who're not too familiar with llvm's builder to visualize, but I can also delete it.